pkg/webhooks/validation_test.go

```diff
@@ -0,0 +1,543 @@
+					"kind": "Pod",
+					"metadata": {
+					   "name": "test-pod",
+					   "labels": {
+						   "app": "my-app"
+					   }
+					},
+					"spec": {
+					   "containers": [
+						  {
+							 "name": "nginx",
+							 "image": "nginx:latest"
+						  }
+					   ]
+					}
+				 }
+			`),
+			blocked: false,
+		},
+		{
+			rawPolicy: []byte(`
+				{
+					"apiVersion": "kyverno.io/v1",
+					"kind": "ClusterPolicy",
+					"metadata": {
+					   "name": "check-label-app"
+					},
+					"spec": {
+					   "validationFailureAction": "enforce",
+					   "validationFailureActionOverrides": 
+							[
+								{
+									"action": "enforce",
+									"namespaces": [
+										"default"
+									]
+								},
+								{
+									"action": "audit",
+									"namespaces": [
+										"test"
+									]
+								}
+							],
+					   "rules": [
+						  {
+							 "name": "check-label-app",
+							 "match": {
+								"resources": {
+								   "kinds": [
+									  "Pod"
+								   ]
+								}
+							 },
+							 "validate": {
+								"message": "The label 'app' is required.",
+								"pattern": {
+									"metadata": {
+										"labels": {
+											"app": "?*"
+										}
+									}
+								}
+							}
+						  }
+					   ]
+					}
+				 }
+			`),
+			rawResource: []byte(`
+				{
+					"apiVersion": "v1",
+					"kind": "Pod",
+					"metadata": {
+					   "name": "test-pod",
+					   "namespace": "test"
+					},
+					"spec": {
+					   "containers": [
+						  {
+							 "name": "nginx",
+							 "image": "nginx:latest"
+						  }
+					   ]
+					}
+				 }
+			`),
+			blocked: false,
+		},
+		{
+			rawPolicy: []byte(`
+				{
+					"apiVersion": "kyverno.io/v1",
+					"kind": "ClusterPolicy",
+					"metadata": {
+					   "name": "check-label-app"
+					},
+					"spec": {
+					   "validationFailureAction": "enforce",
+					   "validationFailureActionOverrides": 
+							[
+								{
+									"action": "enforce",
+									"namespaces": [
+										"default"
+									]
+								},
+								{
+									"action": "audit",
+									"namespaces": [
+										"test"
+									]
+								}
+							],
+					   "rules": [
+						  {
+							 "name": "check-label-app",
+							 "match": {
+								"resources": {
+								   "kinds": [
+									  "Pod"
+								   ]
+								}
+							 },
+							 "validate": {
+								"message": "The label 'app' is required.",
+								"pattern": {
+									"metadata": {
+										"labels": {
+											"app": "?*"
+										}
+									}
+								}
+							}
+						  }
+					   ]
+					}
+				 }
+			`),
+			rawResource: []byte(`
+				{
+					"apiVersion": "v1",
+					"kind": "Pod",
+					"metadata": {
+					   "name": "test-pod",
+					   "namespace": ""
+					},
+					"spec": {
+					   "containers": [
+						  {
+							 "name": "nginx",
+							 "image": "nginx:latest"
+						  }
+					   ]
+					}
+				 }
+			`),
+			blocked: true,
+		},
+	}
+
+	for i, tc := range testcases {
```

Comment:
Should we add 1 more test case to verify the behavior, where  both enforce and audit has the same namespace mentioned by mistake. Are we going to say its a invalid policy or ?

```json
					   "validationFailureAction": "enforce",
					   "validationFailureActionOverrides": 
							[
								{
									"action": "enforce",
									"namespaces": [
										"test"
									]
								},
								{
									"action": "audit",
									"namespaces": [
										"test"

```

Reply:
Right now, whichever comes first is used. So here, it'd be `enforce` but if we reverse the order, it'd be `audit`.

Should we make invalidate this?

Reply:
Ideally we should, so the user knows the mistake in front and ractify that, instead of we assume it.